adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -664,8 +659,8 @@ private void doUploadMessage(
                                 contentType,
                                 payload,
                                 props,
-                                currentSpan.context()),
-                        responseReady.future())
+                                currentSpan.context()).onFailure(thr -> responseReadyTracker.cancel("send event failed")),
+                        responseReadyTracker.waitForCommand())
```

Comment:
Reading the method name "waitForCommand" here, one could wonder "why does it have to wait for a command even in cases where no `hono-ttd` was set?".
Therefore I would rather use something like "waitForCommandIfNeeded".

Reply:
That was the approach in old code createCommandConsume could or may could not create command consume.
More than that - the caller duUploadMessage had to consider if the consumer is `null` or not
However, I think to rename method to ResponseReadyTracker.waitFor which would hide command aspect comepletely

Reply:
If leaving out the "command" term here, how about using `responseReadyTracker.future()` instead of `waitFor`?
(The term "wait" not being very common when dealing with Futures.)

Reply:
here `waitFor` doesn't mean synchronous wait but I agree that it could be misunderstand. 
I'm ok with `featue` and will change it.